src/main/java/br/com/pagbanks/projeto_onboarding/dto/CartDto.java

```diff
@@ -0,0 +1,35 @@
+package br.com.pagbanks.projeto_onboarding.dto;
+
+import br.com.pagbanks.projeto_onboarding.entity.Cart;
+import br.com.pagbanks.projeto_onboarding.entity.Item;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+
+public record CartDto(
+
+        Long id,
+        @JsonProperty("list_items")
+        Set<Item> listItens,
+        @JsonProperty("creation_date")
+        LocalDate creationDate,
+        @JsonProperty("total_value")
+        Double totalValue
+
+) {
+
+        public CartDto {
+                if (creationDate == null) {
+                        creationDate = LocalDate.now();
+                }
+                if (totalValue == null) {
+                        totalValue = 0.0;
+                }
+        }
+
+        public CartDto(Cart cart){
+                this(cart.getId(),cart.getListItens(),cart.getCreationDate(),cart.getTotalValue());
+        }
```

Comment:
Esse tipo de método que faz conversão é comumente chamado de mapper.
É possível utilizar frameworks como mapstruct para fazer mapeamento, mas também usar o java puro.
Ao invés de um construtor, o mais comum é um método estático com a nomenclatura 'from" ou 'to'. Exemplo nesse caso:
![image](https://github.com/user-attachments/assets/465e270b-1623-4f7b-bb51-d19745af80c4)

É estático para não poder ser chamado independente de instanciação da classe.

A leitura fica fluida:
É um CartDto from Cart.

![image](https://github.com/user-attachments/assets/15bb3bc5-0b4b-4cdd-85fa-d4c19a91c0e6)

Em uma classe focada na conversão, poderia ficar assim:
![image](https://github.com/user-attachments/assets/f7cbd1bf-3e5d-41d0-a4f2-a508cf70f1d7)


Essa necessidade de mapeamento é tão comum que temos frameworks como o Mapstruct que facilitam essas conversões.